crates/holochain_metrics/src/lib.rs

```diff
@@ -0,0 +1,29 @@
+const DASH_NETWORK_STATS: &[u8] = include_bytes!("dashboards/networkstats.json");
+
+/// Initialize metrics. Call this from binaries, not libraries.
+pub async fn initialize_metrics(path: std::path::PathBuf) {
```

Comment:
beyond the initial demo i think it's worth building modules and a metrics config that controls the behavior of this.

e.g.
* just writing otel metrics out somewhere
* set up config for (or even spin up) influxdb and push metrics to it
* etc.

Reply:
@steveej - perhaps it was not clear in the demo video, but this is already doing all this.

It runs InfluxDB as a child process of holochain. Our modules can use the otel api to report metrics, and this config makes those otel metrics be shipped to influx DB.

Reply:
This is why it's in the holochain_metrics crate. That way, if you want something else to do it, you can `disable_metrics` in the holochain config. I don't understand why we'd want to *not* include it as an option since it's easy to do.

We *do* need to make the metrics config a little more comprehensive of course.

Reply:
> I don't understand why we'd want to _not_ include it as an option since it's easy to do.

i'm for including it but in the following way:

configuration allows to..
* opt-out of writing metrics to the filesystem
* opt-in to running InfluxDB configured with an informative dashboard and consuming the metrics from the default location

Reply:
I feel like there's a misunderstanding here. InfluxDB is the thing writing metrics to the filesystem. So... if we want them to be written in InfluxDBs database, we need influxdb present. We *can* disable the UI server, but it doesn't use any significant resources when no browser is open to it, so I don't much see the point of disabling it.

Reply:
the metrics that InfluxDB writes to the filesystem are going to be specific to itself. if those are the only metrics we store it seems to be defeating the purpose of using OpenTelemetry as a standard.

the suggested approach is to use the [OpenTelemetry Collector](https://opentelemetry.io/docs/collector/) as an intermediary component.
this yields the following architecture

* holochain instruments the [OTLP exporter](https://crates.io/crates/opentelemetry-otlp] to push to the OTLC
* OTLC is configured with the [InfluxDB exporter](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/exporter/influxdbexporter)
* influxdb is configured with the [OTLP receiver](https://github.com/influxdata/telegraf/blob/release-1.21/plugins/inputs/opentelemetry/README.md)

Reply:
@steveej the collector doesn't actually collect anything, it just forwards to something else, so we'd have to run *another* binary on the system just to put it in to influx anyways.

Using opentelemetry is *not* defeated. It allows us to, at config time, switch it to any collection system we want to use.

For starters we will only have influx, but we can write anything we want at any point. But I'd submit that all the utility we actually *want* out of a metrics system is actually encompassed by influx, so let's start with this and see how far it gets us.

Reply:
if we start out directly pushing to the OTLC that's the only component holochain will ever push observability data to. we can start having influxdb bei the only consumer on the other side of the OTLC.

if i think beyond a single conductor scenario, and especially if we consume data from multiple distributed conductors, it seems beneficial if holochain pushes to the OTLC process that lives on the same host, and that process then takes care of either being pulled or push data cross the localhost boundary.

i'm happy to help out with the spinning up of the OTLC if you want to pair up on this.

Reply:
Otel already has otlc push capabilities, so if we want to do that, we can do that easily. But in a launcher scenario, when influx is the goal, why would we want to have to have the intermediary?

You want to be able to push metrics to otlc. Fine, let's build in the conductor config that lets that happen. Why would we not make use of the ability to push directly to influxdb, when that is how we want to configure it, though?

Reply:
Okay, in the PR, metrics are now disabled by default.

- If you want to enabled them with child process, set the env var `HOLOCHAIN_INFLUXIVE_CHILD_SVC=1`

---

- If you want to enable them for an already running influxd:
  - `HOLOCHAIN_INFLUXIVE_EXTERNAL=1`
  - `HOLOCHAIN_INFLUXIVE_EXTERNAL_HOST=[your host url]`
  - `HOLOCHAIN_INFLUXIVE_EXTERNAL_BUCKET=[your bucket name]`
  - `HOLOCHAIN_INFLUXIVE_EXTERNAL_TOKEN=[your auth token]`

And, obviously, it will ***not*** be able to pre-configure any dashboards for you.

Reply:
disabling this by default gives us a chance to iterate on this by keeping the experiments to those who opt-in :+1: 

configuration, including preconfigured dashboards, can be made reusable as well so i'm not worried about that.